client/components/accordion/index.jsx

```diff
@@ -50,15 +50,21 @@ export default class Accordion extends Component {
 		this.props.onToggle( isExpanded );
 	};
 
+	_mountChildren = false;
+
 	render() {
 		const { className, icon, title, subtitle, status, children, e2eTitle } = this.props;
+		const isExpanded = this.state.isExpanded || this.props.forceExpand;
 		const classes = classNames( 'accordion', className, {
-			'is-expanded': this.state.isExpanded || this.props.forceExpand,
+			'is-expanded': isExpanded,
 			'has-icon': !! icon,
 			'has-subtitle': !! subtitle,
 			'has-status': !! status,
 		} );
 
+		// Keep children off the render tree until it's first expanded.
+		this._mountChildren = this._mountChildren || isExpanded;
```

Comment:
I may not get the full point here but isn't this something we should keep in component state?

Reply:
I don't think there's any particular problem in keeping it as an instance variable. And as @jsnajdr mentioned, it would add some complexity.

My rule of thumb is to only add things to state that should trigger a new render, at least in some scenario. But in this case, this is a boolean that we only want to check (and set) during render, without the need for it ever to trigger a new render cycle. The only downside I see to this approach is some added difficulty in debugging, perhaps, as it's harder to get a snapshot of everything that's going on inside the component.

I'm still somewhat new to React, though, so my rule of thumb may be woefully inadequate :)

Reply:
I've no strong opinion here. Thanks for clarification @jsnajdr and @sgomes, this is good for me :+1: